pandas/tests/resample/test_datetime_index.py

```diff
@@ -25,1484 +23,1491 @@
 
-    def test_resample_timedelta_idempotency(self):
+def test_resample_dup_index():
 
-        # GH 12072
-        index = pd.timedelta_range('0', periods=9, freq='10L')
-        series = Series(range(9), index=index)
-        result = series.resample('10L').mean()
-        expected = series
-        assert_series_equal(result, expected)
+    # GH 4812
+    # dup columns with resample raising
+    df = DataFrame(np.random.randn(4, 12), index=[2000, 2000, 2000, 2000],
+                   columns=[Period(year=2000, month=i + 1, freq='M')
+                            for i in range(12)])
+    df.iloc[3, :] = np.nan
+    result = df.resample('Q', axis=1).mean()
+    expected = df.groupby(lambda x: int((x.month - 1) / 3), axis=1).mean()
+    expected.columns = [
+        Period(year=2000, quarter=i + 1, freq='Q') for i in range(4)]
+    assert_frame_equal(result, expected)
+
+
+def test_resample_reresample():
+    dti = date_range(start=datetime(2005, 1, 1),
+                     end=datetime(2005, 1, 10), freq='D')
+    s = Series(np.random.rand(len(dti)), dti)
+    bs = s.resample('B', closed='right', label='right').mean()
+    result = bs.resample('8H').mean()
+    assert len(result) == 22
+    assert isinstance(result.index.freq, offsets.DateOffset)
+    assert result.index.freq == offsets.Hour(8)
+
+
+def test_resample_timestamp_to_period(simple_date_range_series):
+    ts = simple_date_range_series('1/1/1990', '1/1/2000')
+
+    result = ts.resample('A-DEC', kind='period').mean()
+    expected = ts.resample('A-DEC').mean()
+    expected.index = period_range('1990', '2000', freq='a-dec')
+    assert_series_equal(result, expected)
+
+    result = ts.resample('A-JUN', kind='period').mean()
+    expected = ts.resample('A-JUN').mean()
+    expected.index = period_range('1990', '2000', freq='a-jun')
+    assert_series_equal(result, expected)
+
+    result = ts.resample('M', kind='period').mean()
+    expected = ts.resample('M').mean()
+    expected.index = period_range('1990-01', '2000-01', freq='M')
+    assert_series_equal(result, expected)
+
+    result = ts.resample('M', kind='period').mean()
+    expected = ts.resample('M').mean()
+    expected.index = period_range('1990-01', '2000-01', freq='M')
+    assert_series_equal(result, expected)
+
+
+def test_ohlc_5min():
+    def _ohlc(group):
+        if isna(group).all():
+            return np.repeat(np.nan, 4)
+        return [group[0], group.max(), group.min(), group[-1]]
+
+    rng = date_range('1/1/2000 00:00:00', '1/1/2000 5:59:50', freq='10s')
+    ts = Series(np.random.randn(len(rng)), index=rng)
+
+    resampled = ts.resample('5min', closed='right',
+                            label='right').ohlc()
+
+    assert (resampled.loc['1/1/2000 00:00'] == ts[0]).all()
+
+    exp = _ohlc(ts[1:31])
+    assert (resampled.loc['1/1/2000 00:05'] == exp).all()
+
+    exp = _ohlc(ts['1/1/2000 5:55:01':])
+    assert (resampled.loc['1/1/2000 6:00:00'] == exp).all()
+
+
+def test_downsample_non_unique():
```

Comment:
on these extra long files, ok to put in some section separators and move functions around

somthing like
```
# ----
# ohlc
# ----
```
these make sense when we remove the class based testing, not super necessary but nice to have.

Reply:
can you address this

Reply:
There's still lots to do for this module... only a handful of tests were split or parametrised in this PR. i would prefer to move functions and add section separators as a parallel activity with the splitting and parametrizing of the remaining tests. are you OK with that?